### DISCUSSION TITLE
🙏 Streamline & retire unnecessary prompts in the Yeoman generator SPFx 1.14+

### DISCUSSION THREAD

**Discussion by andrewconnell** [SharePoint/sp-dev-docs#7585] [General]
Some of the prompts in the generator have been retired. Some of the long-standing prompts in the generator are long in the tooth and hard to understand for new users.

Recommendation: remove these confusing/retired prompts & instead, point to a page in the official docs that provide better explanations & guidance

## TL/DR;

The generator should be used to create new projects & shouldn't be used to provide documentation or explanations; that's what official DOCS are for.

### Why

Docs are better for this because of the following reasons:

- **Provide more context**: Docs can be much more wordy and provide a more explanation as to why something is done a certain way. The console is not a great place to read long sentences of explanations.
- **Include rich content**: Docs can be formatted with rich content, including links, emphasis, notes, callouts, code examples, and images. None of this is possible within the console where Yeoman generators are run.
- **Easier to maintain**: Docs are easier to maintain than the Yeoman generator. The only way to update the generator is to release a new version. This usually implies to consumers there's a new release with new/changed features. Docs on the other hand, can be maintained separately from the generator.

### Current state

Take the SPFx v1.13.1 generator... when creating a web part with all the default options, you get the following (*this almost mirrors the v1.14.0.beta4 version as well*):

```console
yo @microsoft/sharepoint --skip-install

     _-----_     ╭──────────────────────────╮
    |       |    │      Welcome to the      │
    |--(o)--|    │  SharePoint Client-side  │
   `---------´   │         Solution         │
    ( _´U`_ )    │     Generator@1.13.1     │
    /___A___\   /╰──────────────────────────╯
     |  ~  |     
   __'.___.'__   
 ´   `  |° ´ Y ` 

Let's create a new SharePoint solution.
? What is your solution name? spfx-1-13-1
? Only SharePoint Online (latest) is supported.  For earlier versions of SharePoint (2016 and 2019) please use the 1.4.1 version of the generator. SharePoint
 Online only (latest)
? Do you want to allow the tenant admin the choice of being able to deploy the solution to all sites immediately without running any feature deployment or ad
ding apps in sites? No
? Will the components in the solution require permissions to access web APIs that are unique and not shared with other components in the tenant? No
? Which type of client-side component to create? WebPart
Add new Web part to solution spfx-1-13-1.
? What is your Web part name? HelloWorld
? What is your Web part description? HelloWorld description
? Which framework would you like to use? No JavaScript framework
```

### Proposal

IMHO, questions 2-4 in the generator are no longer necessary to include. I propose SPFx v1.14.[GA] be the following:

```console
yo @microsoft/sharepoint --skip-install

     _-----_     ╭──────────────────────────────────╮
    |       |    │ Welcome to the SharePoint Online │
   `---------´   │   SPFx Solution Yeoman generator │
    ( _´U`_ )    │          v1.14.0-beta.4          │
    /___A___\   /╰──────────────────────────────────╯
     |  ~  |     
   __'.___.'__   
 ´   `  |° ´ Y ` 

 NOTE: Some of the following options have been retired in favor of their defaults. For
       more information & how to change these default values, see: https://aka.ms/spfx-its-2022

Let's create a new SharePoint solution.
? What is your solution name? spfx-1-14-b-4
? Enter a short description for your solution? spfx-1-14-b-4 description
? Which type of client-side component to create? WebPart
Add new Web part to solution spfx-1-14-b-4.
? What is your Web part name? HelloWorld
? What is your Web part description? HelloWorld description
? Which template would you like to use? Minimal
```

Take note of the following:

1. The title box clearly calls this out as the **SharePoint Online** generator for **SPFx** projects. There's no mention of SPFx anywhere in the generator when that's what you're creating. This has always seemed odd to me.
1. Added a **NOTE** before the prompts that can direct people to the docs for more information to address changes & potential questions of "what happened to the old prompt X" that I'm about to suggest removing.
1. Question #Q2 WRT the target environment... remove this... why say "this is for X, not for Y & Z" and force people to accept it. Just retire it... we can cover this in the docs in the aforementioned docs:

    - starting with SPFx v1.13.x, the generator is only for SPO projects
    - if you want to create SPFx projects for use with SP 2016|2019|SE, point to a page in the docs on what they need to do

1. Questions #Q3 & #Q4 - these have always been long & wordy and difficult for new users to understand what they mean. Personally, I've always had to read & study both of these to wonder "*uh... what are they asking... what are they impacting with my project*".

    Why not, instead, pick the default options these two questions ask in the scaffolded projects. Then, in the docs, explain the following:

    - what the old question was (for SEO)
    - what they did (*ie: set property X to TRUE in file yomama.json*)
    - what the impact is of the default option
    - why were they removed
    - how you can change them
    - include links to the docs for the relevant properties (don't dupe the docs on this generator page... use the generator page I propose to point to the docs for single-source-of-truth)

1. optionally, when the generator has finished creating the project, repeat the "NOTE".

**Comment by wilecoyotegenius** [comment]
I would drop the question about web part/extension description as well. I hardly ever have a good description ready at the time of running the generator. So in my case, project generation usualy ends up with an empty description, often never to be filled in.

**Reply by robpearmain** [reply]
I agree, I never really use the description

**Comment by sparkitect** [comment]
Great idea, @andrewconnell!

Perhaps 

> "Some of the following options have been retired in favor of their defaults." 

could say 

> "Some generator options have been retired in favor of their defaults."

since they are not actually following :)
